libcontainer/cgroups/fs/cpuset_test.go

```diff
@@ -65,3 +81,382 @@ func TestCpusetSetMems(t *testing.T) {
+		"cpuset.sched_relax_domain_level": schedRelaxDomainLevel,
+	})
+
+	cpuset := &CpusetGroup{}
+	actualStats := *cgroups.NewStats()
+	err := cpuset.GetStats(helper.CgroupPath, &actualStats)
+	if err == nil {
+		t.Fatal("Expected failure")
+	}
+}
+func TestCpusetStatsNoMemorySpreadSlabFile(t *testing.T) {
+	helper := NewCgroupTestUtil("cpuset", t)
+	defer helper.cleanup()
+	helper.writeFileContents(map[string]string{
+		"cpuset.cpus":                     cpus,
+		"cpuset.cpu_exclusive":            cpuExclusive,
+		"cpuset.mems":                     mems,
+		"cpuset.mem_hardwall":             memHardwall,
+		"cpuset.mem_exclusive":            memExclusive,
+		"cpuset.memory_migrate":           memoryMigrate,
+		"cpuset.memory_spread_page":       memorySpreadPage,
+		"cpuset.memory_pressure":          memoryPressure,
+		"cpuset.sched_load_balance":       schedLoadBalance,
+		"cpuset.sched_relax_domain_level": schedRelaxDomainLevel,
+	})
+
+	cpuset := &CpusetGroup{}
+	actualStats := *cgroups.NewStats()
+	err := cpuset.GetStats(helper.CgroupPath, &actualStats)
+	if err == nil {
+		t.Fatal("Expected failure")
+	}
+}
+func TestCpusetStatsNoMemoryPressureFile(t *testing.T) {
+	helper := NewCgroupTestUtil("cpuset", t)
+	defer helper.cleanup()
+	helper.writeFileContents(map[string]string{
+		"cpuset.cpus":                     cpus,
+		"cpuset.cpu_exclusive":            cpuExclusive,
+		"cpuset.mems":                     mems,
+		"cpuset.mem_hardwall":             memHardwall,
+		"cpuset.mem_exclusive":            memExclusive,
+		"cpuset.memory_migrate":           memoryMigrate,
+		"cpuset.memory_spread_page":       memorySpreadPage,
+		"cpuset.memory_spread_slab":       memorySpeadSlab,
+		"cpuset.sched_load_balance":       schedLoadBalance,
+		"cpuset.sched_relax_domain_level": schedRelaxDomainLevel,
+	})
+
+	cpuset := &CpusetGroup{}
+	actualStats := *cgroups.NewStats()
+	err := cpuset.GetStats(helper.CgroupPath, &actualStats)
+	if err == nil {
+		t.Fatal("Expected failure")
+	}
+}
+
+func TestCpusetStatsNosSchedLoadBalanceFile(t *testing.T) {
+	helper := NewCgroupTestUtil("cpuset", t)
+	defer helper.cleanup()
+	helper.writeFileContents(map[string]string{
+		"cpuset.cpus":                     cpus,
+		"cpuset.cpu_exclusive":            cpuExclusive,
+		"cpuset.mems":                     mems,
+		"cpuset.mem_hardwall":             memHardwall,
+		"cpuset.mem_exclusive":            memExclusive,
+		"cpuset.memory_migrate":           memoryMigrate,
+		"cpuset.memory_spread_page":       memorySpreadPage,
+		"cpuset.memory_spread_slab":       memorySpeadSlab,
+		"cpuset.memory_pressure":          memoryPressure,
+		"cpuset.sched_relax_domain_level": schedRelaxDomainLevel,
+	})
+
+	cpuset := &CpusetGroup{}
+	actualStats := *cgroups.NewStats()
+	err := cpuset.GetStats(helper.CgroupPath, &actualStats)
+	if err == nil {
+		t.Fatal("Expected failure")
+	}
+}
+func TestCpusetStatsNoSchedRelaxDomainLevelFile(t *testing.T) {
```

Comment:
Perhaps you can combine all these tests into one, by having an initial set of files written, and checking if the values read are as expected, and then checking all the error cases like this
```go
        testCases := []struct {
                desc string
                filename, contents string
                removeFilename string
        }{
                "missing SchedRelaxDomainLevelFile",
                removeFilename = "cpuset.sched_relax_domain_level"
        }

for _, tc := range errTestCases {
￼	cpuset := &CpusetGroup{}
￼	actualStats := *cgroups.NewStats()
￼	err := cpuset.GetStats(helper.CgroupPath, &actualStats)
￼	if err == nil {
￼		t.Errorf("case %s: expected failure, got nil", tc.desc)
￼	}
}
```

Reply:
Condensed them into two tests: positive case and all negative cases